File: scripts/generate_apple_secret.rb
#!/usr/bin/env ruby
# Generate Apple Sign In Client Secret for Supabase
# Usage: ruby generate_apple_secret.rb /path/to/AuthKey_H49QU2J6LU.p8

require 'jwt'
require 'openssl'

# Your Apple Developer Configuration
TEAM_ID = "3SPV37F368"        # Your Apple Team ID
CLIENT_ID = "com.silni.app"   # Your App Bundle ID
KEY_ID = "H49QU2J6LU"         # Your Key ID

# Check if key file path is provided
if ARGV.empty?
  puts "Usage: ruby generate_apple_secret.rb /path/to/AuthKey_H49QU2J6LU.p8"
  puts ""
  puts "Download your .p8 key file from:"
  puts "https://developer.apple.com/account/resources/authkeys/list"
  exit 1
end

key_file = ARGV[0]

unless File.exist?(key_file)
  puts "Error: Key file not found: #{key_file}"
  exit 1
end

# Read the private key
key_content = File.read(key_file)
ecdsa_key = OpenSSL::PKey::EC.new(key_content)

# JWT headers
headers = {
  "kid" => KEY_ID,
  "alg" => "ES256"
}

# JWT claims - valid for 6 months (maximum allowed)
now = Time.now.to_i
claims = {
  "iss" => TEAM_ID,
  "iat" => now,
  "exp" => now + (86400 * 180),  # 6 months
  "aud" => "https://appleid.apple.com",
  "sub" => CLIENT_ID
}

# Generate the JWT
token = JWT.encode(claims, ecdsa_key, "ES256", headers)

puts ""
puts "=" * 60
puts "Apple Sign In Client Secret Generated Successfully!"
puts "=" * 60
puts ""
puts "Copy this entire token and paste it in Supabase:"
puts "Dashboard > Authentication > Providers > Apple > Secret Key"
puts ""
puts "-" * 60
puts token
puts "-" * 60
puts ""
puts "This secret is valid for 6 months."
puts "Regenerate before: #{Time.at(now + 86400 * 180)}"
puts ""

File: scripts/generate_apple_secret.js
#!/usr/bin/env node
/**
 * Generate Apple Sign In Client Secret for Supabase
 *
 * Usage:
 *   npm install jsonwebtoken
 *   node generate_apple_secret.js /path/to/AuthKey_H49QU2J6LU.p8
 */

const fs = require('fs');
const jwt = require('jsonwebtoken');

// Your Apple Developer Configuration
const TEAM_ID = "3SPV37F368";        // Your Apple Team ID
const CLIENT_ID = "com.silni.app";   // Your App Bundle ID
const KEY_ID = "H49QU2J6LU";         // Your Key ID

// Check if key file path is provided
if (process.argv.length < 3) {
  console.log("Usage: node generate_apple_secret.js /path/to/AuthKey_H49QU2J6LU.p8");
  console.log("");
  console.log("Download your .p8 key file from:");
  console.log("https://developer.apple.com/account/resources/authkeys/list");
  process.exit(1);
}

const keyFile = process.argv[2];

// Check if file exists
if (!fs.existsSync(keyFile)) {
  console.log(`Error: Key file not found: ${keyFile}`);
  process.exit(1);
}

// Read the private key
const privateKey = fs.readFileSync(keyFile, 'utf8');

// Current time
const now = Math.floor(Date.now() / 1000);

// JWT claims - valid for 6 months (maximum allowed by Apple)
const claims = {
  iss: TEAM_ID,
  iat: now,
  exp: now + (86400 * 180),  // 6 months
  aud: "https://appleid.apple.com",
  sub: CLIENT_ID
};

// Generate the JWT
const token = jwt.sign(claims, privateKey, {
  algorithm: 'ES256',
  header: {
    kid: KEY_ID,
    alg: 'ES256'
  }
});

console.log("");
console.log("=".repeat(60));
console.log("Apple Sign In Client Secret Generated Successfully!");
console.log("=".repeat(60));
console.log("");
console.log("Copy this entire token and paste it in Supabase:");
console.log("Dashboard > Authentication > Providers > Apple > Secret Key");
console.log("");
console.log("-".repeat(60));
console.log(token);
console.log("-".repeat(60));
console.log("");
console.log("This secret is valid for 6 months.");
console.log(`Regenerate before: ${new Date((now + 86400 * 180) * 1000).toISOString()}`);
console.log("");
